File: android/app/src/main/java/com/adqmobile/socialnetwork/domain/post/usecases/CreatePostUseCase.kt
package com.adqmobile.socialnetwork.domain.post.usecases

import com.adqmobile.socialnetwork.domain.post.Post
import com.adqmobile.socialnetwork.domain.post.PostRepository

class CreatePostUseCase(private val postRepository: PostRepository) {

    suspend fun run(post: Post) {
        postRepository.createPost(post)
    }
}

File: react-native/domain/post/usecases/CreatePostUseCase.ts
import Post from "../Post";
import PostRepository from "../PostRepository";

export default class CreatePostUseCase {
    postRepository: PostRepository;

    constructor(postRepository: PostRepository) {
        this.postRepository = postRepository;
    }

    async run(post: Post): Promise<Post> {
        return await this.postRepository.createPost(post);
    }
}